src/IssueViz.Security/OpenInIDE/Api/OpenInIDERequestHandler.cs

```diff
@@ -55,16 +80,71 @@ Task IOpenInIDERequestHandler.ShowHotspotAsync(IShowHotspotRequest request)
                 throw new ArgumentNullException(nameof(request));
             }
 
+            return ShowHotspotAsync(request);
+        }
+
+        private async Task ShowHotspotAsync(IShowHotspotRequest request)
+        {
+            logger.WriteLine(OpenInIDEResources.OpenHotspot_ProcessingRequest, request.ServerUrl, request.ProjectKey,
+                request.OrganizationKey ?? OpenInIDEResources.OpenHotspot_NullOrganization, request.HotspotKey);
+
+            // Always show the Hotspots tool window. If we can't successfully process the
+            // request we'll show a gold bar in the window
+            toolWindowService.Show(HotspotsToolWindow.ToolWindowId);
+            failureInfoBar.Clear();
+
             if (!ideStateValidator.CanHandleOpenInIDERequest(request.ServerUrl, request.ProjectKey, request.OrganizationKey))
             {
-                return Task.CompletedTask;
+                // We're assuming the validator will have output an explanantion of why the IDE
+                // isn't in the correct state
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspot = await TryGetHotspotData(request.HotspotKey);
+            if (hotspot == null)
+            {
+                ShowFailureInfoBar();
+                return;
+            }
+
+            var hotspotViz = converter.Convert(hotspot);
+            if (hotspotViz == null)
+            {
+                logger.WriteLine(OpenInIDEResources.OpenHotspot_UnableToConvertHotspotData);
+                ShowFailureInfoBar();
+                return;
+            }
+
+            if (!navigator.TryNavigate(hotspotViz))
+            {
+                logger.WriteLine(OpenInIDEResources.ApiHandler_FailedToNavigateToHotspot, hotspotViz.FilePath, hotspotViz.StartLine);
```

Comment:
LocationNavigator has no logging at the moment, and I'm debating if we should add it, at least for opening the document. For hotspots, it's possible that we couldn't resolve the file path -- and I guess we could guess it based on the fact that this log will output a partial file path, but maybe we should instead do the logging in locationNavigator, and then here just display an infobar. WYDT?
It might be a bit noisy for secondary locations, when code is being modified, so that's why I'm thinking of adding a log only for opening the document.

Reply:
I was expecting we'd have to change the logging once we've started dogfood and can see how it works in practice. I saw there wasn't any logging in the navigator, and for now I just wanted to log something to give us a starting point.